### ISSUE TITLE
cli: hotswap support for AWS::Bedrock::Agent

### ISSUE THREAD

**Issue by tmokmss** [aws/aws-cdk#31296]
### Describe the feature

CDK CLI hotswap should support [`AWS::Bedrock::Agent` resource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-bedrock-agent.html) to quickly deploy the configuration of Agents for Aamzon Bedrock agents.

### Use Case

When we develop an agent of Agents for Amazon Bedrock, we need to deploy CFn template a lot. The definition of an agent contains an instruction, descriptions for available tools, system prompt overrides, etc. Every time we changed these configurations, we have to wait 30~60s to deploy the CFn stack. If we can deploy it with hotswap (usually finishes in a few seconds), it would improve the test cycle greatly.

### Proposed Solution

We can refer to previous pull requests like #27197 as an implementation reference.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.155.0

### Environment details (OS name and version, etc.)

macOS

**Comment by pahud** [comment]
Yes I was recently working on the bedrock agent as well and I agree it would be great to have the hostswap support.

Please help us prioritize with 👍  and we welcome pull requests.
